openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -111,19 +111,22 @@ def extract_dates_from_course(course):
             # unless that item already has a relative date set
             for _, section, weeks_to_complete in spaced_out_sections(course):
                 section_date_items = []
+                max_due_num_weeks = timedelta(weeks=1)
                 for subsection in section.get_children():
                     # If custom pacing is set on a subsection, apply the set relative
                     # date to all the content inside the subsection. Otherwise
                     # apply the default Personalized Learner Schedules (PLS)
                     # logic for self paced courses.
                     due_num_weeks = subsection.fields['due_num_weeks'].read_from(subsection)
                     if (CUSTOM_PLS.is_enabled(course.id) and due_num_weeks):
+                        max_due_num_weeks = max(max_due_num_weeks, timedelta(weeks=due_num_weeks))
                         section_date_items.extend(_get_custom_pacing_children(subsection, due_num_weeks))
                     else:
+                        max_due_num_weeks = max(max_due_num_weeks, weeks_to_complete)
```

Comment:
I believe this isn't necessary because weeks_to_complete comes from spaced_out_sections which already enforces min and max values.

Reply:
Oh wait so this isn't to enforce min and max values but this variable is used to get the maximum number of weeks out of all the subsections